### PR TITLE
Add config for bell

### LINKED ISSUE #125: Allow to disable sounds in configuration file
"Ding" sound for example.

### PR DESCRIPTION
**Pull request by albinekb**
Fix #125

### CODE REVIEW

File: app/config-default.js [hunk 1]
```diff
@@ -54,7 +54,13 @@ module.exports = {
 
     // the shell to run when spawning a new session (i.e. /usr/local/bin/fish)
     // if left empty, your system's login shell will be used by default
-    shell: ''
+    shell: '',
+
+    // URL to custom bell
+    // bellSoundURL: 'http://example.com/bell.mp3',
```
Comment (rauchg) [inline app/config-default.js]: This comment / flag should be bellow `bell`, not before it 👼

File: lib/components/term.js [hunk 2]
```diff
@@ -200,6 +200,12 @@ export default class Term extends Component {
     if (this.props.customCSS !== nextProps.customCSS) {
       this.term.prefs_.set('user-css', this.getStylesheet(nextProps.customCSS));
     }
+
+    if (this.props.bell === 'SOUND') {
+      this.term.prefs_.set('audible-bell-sound', this.props.bellSoundURL);
+    } else {
+      this.term.prefs_.set('audible-bell-sound', '');
+    }
```
Comment (rauchg) [inline lib/components/term.js]: You also need to add it to the `constructor`

File: lib/reducers/ui.js [hunk 3]
```diff
@@ -124,6 +127,14 @@ const reducer = (state = initial, action) => {
           ret.termCSS = config.termCSS;
         }
 
+        if (allowedBells.includes(config.bell)) {
```
Comment (rauchg) [inline lib/reducers/ui.js]: how about `null`. Why is it not a `Set` and `.has` here?

Comment (albinekb) [inline lib/reducers/ui.js]: When would it be `null`? Yeah I'll change it to a `Set` :)

File: lib/components/term.js (hunk 2 above)
Comment (albinekb) [inline lib/components/term.js]: Where?

Comment (rauchg) [inline lib/components/term.js]: Where we initialize the term and its preferences. `componentWillReceiveProps` only gets invoked when `props` change, not on initial render. So the sound pref wouldn't be honored unless you change the config.

Comment (albinekb) [inline lib/components/term.js]: Oh, added that https://github.com/zeit/hyperterm/pull/468/files#diff-268710614c2274f611be667a5f72bb13R18

**Comment by MrRio** [comment]
Should this be true/false rather that 'SOUND' or false?

**Comment by efegurkan** [comment]
@MrRio normally you would have, something like visual flash, bell, beep etc. And you would want them to work together so a boolean value would be much more applicable.

But in case there is other options for sound like beep or some other sound, "SOUND" approach would be better.

**Comment by albinekb** [comment]
There will be 'VISUAL' or 'BELL' aswell :)

**Comment by rauchg** [comment]
LGTM

**Comment by indutny** [comment]
Let's land this :)

**Comment by albinekb** [comment]
Rebased, should really have merged master so that the commits/comments would have stayed 👍  will do that next time.

**Comment by MrRio** [comment]
This works really nicely. Tried with external MP3 URL 👍

**Comment by ericmarkmartin** [comment]
I feel like this should be added to the docs on http://hyper.is.

**Comment by nagman** [comment]
it's not working for me. I put bell: false and I still have this f*** sound every second (I use a lot the escape key).

**Comment by chabou** [comment]
@nagman Please follow this issue https://github.com/zeit/hyper/issues/1613

**Comment by guidobouman** [comment]
Hey @albinekb, what about the `VISUAL` option? :D

**Pull request merged**